src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -78,6 +79,16 @@ public KubernetesSlave(PodTemplate template, String nodeDescription, KubernetesC
         this.cloud = cloud;
     }
 
+    @DataBoundConstructor
+    public KubernetesSlave(PodTemplate template, String nodeDescription, KubernetesCloud cloud, String labelStr,
+                           RetentionStrategy rs)
+            throws Descriptor.FormException, IOException {
+
+        this(template, nodeDescription, cloud, labelStr);
```

Comment:
I think you'd better do it the other way around, call `this(template, nodeDescription, cloud, labelStr, rs);` in the first constructor, call `super` in the second one